### ISSUE TITLE
Upload fails on files larger than 10 MB for LDAP users

### ISSUE THREAD

**Issue by fsutil** [owncloud/core#35239]
This is my first post on Github, so I apologize if I'm posting in the wrong spot, missing info, etc.

I ran into an issue where LDAP users can't upload files larger than 10 MB to external storage. I believe this problem is similar to the one that affected guest accounts (https://github.com/owncloud/guests/issues/198), but I didn't see a reported issue that affects LDAP users. Our scenario is quite similar in that our LDAP users have a 0 B quota, so there's no place for the file "chunking" to take place. The reason for the 0 B quota is so that users can't save files directly to ownCloud, only to our primary file server. This is used solely for remote access to our internal file server. 
 
### Steps to reproduce
1. Map Windows SMB share in External Storage, using Log-in credentials for authentication
2. Set Default quota to 0 B 
3. Login as LDAP user
4. Upload file larger than 10 MB to External SMB share

### Expected behaviour
File should upload successfully or at least give some type of visible error.

### Actual behaviour
File is not uploaded and there is no visible error. If you have Chrome DevTools open, you'll see an error in the console that can be traced back to specific debug logs.

### Workaround I sucessfully tested
from @voroyam - https://github.com/owncloud/guests/issues/198#issuecomment-375272592

1. Create a /tmp/uploads
2. Give the folder to your webserver-user chown -R www-data:www-data /tmp/uploads
3. Add this to config.php 'dav.chunk_base_dir' => '/tmp/uploads',
4. Upload files as LDAP user > 10 MB

### Server configuration
**Operating system:** Ubuntu 18.04

**Web server:** Apache 2.4.29

**Database:** MySQL 5.7.26 (Local)

**PHP version:** 7.2

**ownCloud version:** 10.1.1

**Updated from an older ownCloud or fresh install:** regular incremental updates since 10.0.7 on current Ubuntu 18.04 server. Data migrated from previous Ubuntu 14.04 appliance.

**Where did you install ownCloud from:** ownCloud package repository

**Signing status (ownCloud 9.0 and above):** 

```
No errors have been found.
```

**The content of config/config.php:**

```
{
    "system": {
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "localhost",
            "***FQDN***",
            "***IP***"
        ],
        "datadirectory": "\/var\/www\/owncloud\/data",
        "overwrite.cli.url": "http:\/\/localhost",
        "dbtype": "mysql",
        "version": "10.1.1.1",
        "dbname": "oc",
        "dbhost": "localhost",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "logtimezone": "UTC",
        "installed": true,
        "memcache.local": "\\OC\\Memcache\\APCu",
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "localhost",
            "port": 6379
        },
        "ldapIgnoreNamingRules": false,
        "theme": "theme1",
        "instanceid": "oc5go2jf83je",
        "maintenance": false,
        "loglevel": 2,
        "skeletondirectory": "",
        "updatechecker": false,
        "smb.logging.enable": true,
        "dav.chunk_base_dir": "\/tmp\/uploads"
    }
}

```

**List of activated apps:**

```
Enabled:
  - configreport: 0.1.2
  - dav: 0.4.0
  - diagnostics: 0.1.4
  - federatedfilesharing: 0.3.1
  - files: 1.5.2
  - files_external: 0.7.1
  - files_pdfviewer: 0.11.0
  - files_videoplayer: 0.9.9
  - gallery: 16.1.1
  - market: 0.4.0
  - notifications: 0.4.1
  - templateeditor: 0.4.0
  - updatenotification: 0.2.1
  - user_ldap: 0.13.0
Disabled:
  - comments
  - encryption
  - external
  - federation
  - files_sharing
  - files_textviewer
  - files_trashbin
  - files_versions
  - firstrunwizard
  - provisioning_api
  - systemtags
  - user_external

```

**Are you using external storage, if yes which one:**  smb

**Are you using encryption:** no

**Are you using an external user-backend, if yes which one:** LDAP

#### LDAP configuration 

```
+-------------------------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Configuration                 |                                                                                                                                                                                                                         |
+-------------------------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| hasMemberOfFilterSupport      | 1                                                                                                                                                                                                                       |
| hasPagedResultSupport         |                                                                                                                                                                                                                         |
| homeFolderNamingRule          |                                                                                                                                                                                                                         |
| lastJpegPhotoLookup           | 0                                                                                                                                                                                                                       |
| ldapAgentName                 | CN=ownCloud LDAP,OU=***REMOVED SENSITIVE VALUE***,DC=domain,DC=com                                                                                                                                                      |
| ldapAgentPassword             | ***                                                                                                                                                                                                                     |
| ldapAttributesForGroupSearch  |                                                                                                                                                                                                                         |
| ldapAttributesForUserSearch   |                                                                                                                                                                                                                         |
| ldapBackupHost                |                                                                                                                                                                                                                         |
| ldapBackupPort                |                                                                                                                                                                                                                         |
| ldapBase                      | DC=domain,DC=com                                                                                                                                                                                                        |
| ldapBaseGroups                | DC=domain,DC=com                                                                                                                                                                                                        |
| ldapBaseUsers                 | DC=domain,DC=com                                                                                                                                                                                                        |
| ldapCacheTTL                  | 600                                                                                                                                                                                                                     |
| ldapConfigurationActive       | 1                                                                                                                                                                                                                       |
| ldapDynamicGroupMemberURL     |                                                                                                                                                                                                                         |
| ldapEmailAttribute            | mail                                                                                                                                                                                                                    |
| ldapExperiencedAdmin          | 0                                                                                                                                                                                                                       |
| ldapExpertUUIDGroupAttr       |                                                                                                                                                                                                                         |
| ldapExpertUUIDUserAttr        | sAMAccountName                                                                                                                                                                                                          |
| ldapExpertUsernameAttr        |                                                                                                                                                                                                                         |
| ldapGroupDisplayName          | cn                                                                                                                                                                                                                      |
| ldapGroupFilter               | (&(objectclass=group)(|(memberOf:1.2.840.113556.1.4.1941:=CN=UserGroup1,DC=domain,DC=com)(cn=UserGroup1)(memberOf:1.2.840.113556.1.4.1941:=CN=UserGroup2,OU=***REMOVED SENSITIVE VALUE***,DC=domain,DC=com)(cn=UserGroup2))) |
| ldapGroupFilterGroups         |                                                                                                                                                                                                                         |
| ldapGroupFilterMode           | 1                                                                                                                                                                                                                       |
| ldapGroupFilterObjectclass    |                                                                                                                                                                                                                         |
| ldapGroupMemberAssocAttr      | member                           +-------------------------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Configuration                 |                                                                                                                                                                                                                         |
+-------------------------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| hasMemberOfFilterSupport      | 1                                                                                                                                                                                                                       |
| hasPagedResultSupport         |                                                                                                                                                                                                                         |
| homeFolderNamingRule          |                                                                                                                                                                                                                         |
| lastJpegPhotoLookup           | 0                                                                                                                                                                                                                       |
| ldapAgentName                 | CN=ownCloud LDAP,OU=***REMOVED SENSITIVE VALUE***,DC=domain,DC=com                                                                                                                                                      |
| ldapAgentPassword             | ***                                                                                                                                                                                                                     |
| ldapAttributesForGroupSearch  |                                                                                                                                                                                                                         |
| ldapAttributesForUserSearch   |                                                                                                                                                                                                                         |
| ldapBackupHost                |                                                                                                                                                                                                                         |
| ldapBackupPort                |                                                                                                                                                                                                                         |
| ldapBase                      | DC=domain,DC=com                                                                                                                                                                                                        |
| ldapBaseGroups                | DC=domain,DC=com                                                                                                                                                                                                        |
| ldapBaseUsers                 | DC=domain,DC=com                                                                                                                                                                                                        |
| ldapCacheTTL                  | 600                                                                                                                                                                                                                     |
| ldapConfigurationActive       | 1                                                                                                                                                                                                                       |
| ldapDynamicGroupMemberURL     |                                                                                                                                                                                                                         |
| ldapEmailAttribute            | mail                                                                                                                                                                                                                    |
| ldapExperiencedAdmin          | 0                                                                                                                                                                                                                       |
| ldapExpertUUIDGroupAttr       |                                                                                                                                                                                                                         |
| ldapExpertUUIDUserAttr        | sAMAccountName                                                                                                                                                                                                          |
| ldapExpertUsernameAttr        |                                                                                                                                                                                                                         |
| ldapGroupDisplayName          | cn                                                                                                                                                                                                                      |
| ldapGroupFilter               | (&(objectclass=group)(|(memberOf:1.2.840.113556.1.4.1941:=CN=UserGroup1,DC=domain,DC=com)(cn=UserGroup1)(memberOf:1.2.840.113556.1.4.1941:=CN=UserGroup2,OU=***REMOVED SENSITIVE VALUE***,DC=domain,DC=com)(cn=UserGroup2))) |
| ldapGroupFilterGroups         |                                                                                                                                                                                                                         |
| ldapGroupFilterMode           | 1                                                                                                                                                                                                                       |
| ldapGroupFilterObjectclass    |                                                                                                                                                                                                                         |
| ldapGroupMemberAssocAttr      | member                                                                                                                                                                                                                  |
| ldapHost                      | ldaps://domain.com                                                                                                                                                                                                      |
| ldapIgnoreNamingRules         |                                                                                                                                                                                                                         |
| ldapLoginFilter               | (&(&(|(objectclass=person))(|(memberOf:1.2.840.113556.1.4.1941:=CN=UserGroup1,DC=domain,DC=com)(cn=UserGroup1)(memberOf:1.2.840.113556.1.4.1941:=CN=UserGroup2,OU=***REMOVED SENSITIVE VALUE***,DC=domain,DC=com)))(samaccountname=%uid)) |
| ldapLoginFilterAttributes     |                                                                                                                                                                                                                         |
| ldapLoginFilterEmail          | 0                                                                                                                                                                                                                       |
| ldapLoginFilterMode           | 1                                                                                                                                                                                                                       |
| ldapLoginFilterUsername       | 1                                                                                                                                                                                                                       |
| ldapNestedGroups              | 1                                                                                                                                                                                                                       |
| ldapOverrideMainServer        |                                                                                                                                                                                                                         |
| ldapPagingSize                | 500                                                                                                                                                                                                                     |
| ldapPort                      | 636                                                                                                                                                                                                                     |
| ldapQuotaAttribute            |                                                                                                                                                                                                                         |
| ldapQuotaDefault              |                                                                                                                                                                                                                         |
| ldapTLS                       | 0                                                                                                                                                                                                                       |
| ldapUserDisplayName           | displayname                                                                                                                                                                                                             |
| ldapUserDisplayName2          |                                                                                                                                                                                                                         |
| ldapUserFilter                | (&(|(objectclass=person))(|(memberOf:1.2.840.113556.1.4.1941:=CN=UserGroup1,DC=domain,DC=com)(cn=UserGroup1)(memberOf:1.2.840.113556.1.4.1941:=CN=UserGroup2,OU=***REMOVED SENSITIVE VALUE***,DC=domain,DC=com)))       |
| ldapUserFilterGroups          |                                                                                                                                                                                                                         |
| ldapUserFilterMode            | 1                                                                                                                                                                                                                       |
| ldapUserFilterObjectclass     | person                                                                                                                                                                                                                  |
| ldapUserName                  | samaccountname                                                                                                                                                                                                          |
| ldapUuidGroupAttribute        | auto                                                                                                                                                                                                                    |
| ldapUuidUserAttribute         | auto                                                                                                                                                                                                                    |
| turnOffCertCheck              | 0                                                                                                                                                                                                                       |
| useMemberOfToDetectMembership | 1                                                                                                                                                                                                                       |
+-------------------------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
                                                                                                                                                                                       |
| ldapHost                      | ldaps://domain.com                                                                                                                                                                                                      |
| ldapIgnoreNamingRules         |                                                                                                                                                                                                                         |
| ldapLoginFilter               | (&(&(|(objectclass=person))(|(memberOf:1.2.840.113556.1.4.1941:=CN=UserGroup1,DC=domain,DC=com)(cn=UserGroup1)(memberOf:1.2.840.113556.1.4.1941:=CN=UserGroup2,OU=***REMOVED SENSITIVE VALUE***,DC=domain,DC=com)))(samaccountname=%uid)) |
| ldapLoginFilterAttributes     |                                                                                                                                                                                                                         |
| ldapLoginFilterEmail          | 0                                                                                                                                                                                                                       |
| ldapLoginFilterMode           | 1                                                                                                                                                                                                                       |
| ldapLoginFilterUsername       | 1                                                                                                                                                                                                                       |
| ldapNestedGroups              | 1                                                                                                                                                                                                                       |
| ldapOverrideMainServer        |                                                                                                                                                                                                                         |
| ldapPagingSize                | 500                                                                                                                                                                                                                     |
| ldapPort                      | 636                                                                                                                                                                                                                     |
| ldapQuotaAttribute            |                                                                                                                                                                                                                         |
| ldapQuotaDefault              |                                                                                                                                                                                                                         |
| ldapTLS                       | 0                                                                                                                                                                                                                       |
| ldapUserDisplayName           | displayname                                                                                                                                                                                                             |
| ldapUserDisplayName2          |                                                                                                                                                                                                                         |
| ldapUserFilter                | (&(|(objectclass=person))(|(memberOf:1.2.840.113556.1.4.1941:=CN=UserGroup1,DC=domain,DC=com)(cn=UserGroup1)(memberOf:1.2.840.113556.1.4.1941:=CN=UserGroup2,OU=***REMOVED SENSITIVE VALUE***,DC=domain,DC=com)))       |
| ldapUserFilterGroups          |                                                                                                                                                                                                                         |
| ldapUserFilterMode            | 1                                                                                                                                                                                                                       |
| ldapUserFilterObjectclass     | person                                                                                                                                                                                                                  |
| ldapUserName                  | samaccountname                                                                                                                                                                                                          |
| ldapUuidGroupAttribute        | auto                                                                                                                                                                                                                    |
| ldapUuidUserAttribute         | auto                                                                                                                                                                                                                    |
| turnOffCertCheck              | 0                                                                                                                                                                                                                       |
| useMemberOfToDetectMembership | 1                                                                                                                                                                                                                       |
+-------------------------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+

```

### Client configuration
**Browser:** Chrome

**Operating system:**
Windows 10 (1809)

#### Portion of ownCloud log  (data/owncloud.log) - with debug entries relating to the error found in Chrome DevTools
```
{"reqId":"VIPWeSHsaEiBwKxZOQlw","level":0,"time":"2019-05-14T18:40:09+00:00","remoteAddr":"192.168.1.2","user":"User1","app":"smb","method":"GET","url":"\/remote.php\/dav\/uploads\/User1\/web-file-upload-901639628a9eda6f95d484a2f666ff5f-1557859208198","message":"enter: __construct({\"domain\":\"domain.com\",\"host\":\"ms-info-3004.domain.com\",\"root\":\"\",\"share\":\"Shared\",\"user\":\"domain.com\\\\User1\",\"password\":\"***removed***\"})"}
{"reqId":"VIPWeSHsaEiBwKxZOQlw","level":0,"time":"2019-05-14T18:40:09+00:00","remoteAddr":"192.168.1.2","user":"User1","app":"smb","method":"GET","url":"\/remote.php\/dav\/uploads\/User1\/web-file-upload-901639628a9eda6f95d484a2f666ff5f-1557859208198","message":"using Icewind\\SMB\\Native\\NativeShare for the connection"}
{"reqId":"VIPWeSHsaEiBwKxZOQlw","level":0,"time":"2019-05-14T18:40:09+00:00","remoteAddr":"192.168.1.2","user":"User1","app":"smb","method":"GET","url":"\/remote.php\/dav\/uploads\/User1\/web-file-upload-901639628a9eda6f95d484a2f666ff5f-1557859208198","message":"leave: __construct, getId:smb::domain.com\\User1@ms-info-3004.domain.com\/\/Shared\/\/"}
{"reqId":"VIPWeSHsaEiBwKxZOQlw","level":0,"time":"2019-05-14T18:40:09+00:00","remoteAddr":"192.168.1.2","user":"User1","app":"smb","method":"GET","url":"\/remote.php\/dav\/uploads\/User1\/web-file-upload-901639628a9eda6f95d484a2f666ff5f-1557859208198","message":"enter: __construct({\"host\":\"fileserver.domain.com\",\"share\":\"Users\",\"root\":\"UserGroup1\\\\User1\",\"domain\":\"domain.com\",\"user\":\"domain.com\\\\User1\",\"password\":\"***removed***\"})"}
{"reqId":"VIPWeSHsaEiBwKxZOQlw","level":0,"time":"2019-05-14T18:40:09+00:00","remoteAddr":"192.168.1.2","user":"User1","app":"smb","method":"GET","url":"\/remote.php\/dav\/uploads\/User1\/web-file-upload-901639628a9eda6f95d484a2f666ff5f-1557859208198","message":"using Icewind\\SMB\\Native\\NativeShare for the connection"}
{"reqId":"VIPWeSHsaEiBwKxZOQlw","level":0,"time":"2019-05-14T18:40:09+00:00","remoteAddr":"192.168.1.2","user":"User1","app":"smb","method":"GET","url":"\/remote.php\/dav\/uploads\/User1\/web-file-upload-901639628a9eda6f95d484a2f666ff5f-1557859208198","message":"leave: __construct, getId:smb::domain.com\\User1@fileserver.domain.com\/\/Users\/\/UserGroup1\\User1\/"}
{"reqId":"VIPWeSHsaEiBwKxZOQlw","level":0,"time":"2019-05-14T18:40:09+00:00","remoteAddr":"192.168.1.2","user":"User1","app":"smb","method":"GET","url":"\/remote.php\/dav\/uploads\/User1\/web-file-upload-901639628a9eda6f95d484a2f666ff5f-1557859208198","message":"enter: __construct({\"host\":\"fileserver.domain.com\",\"share\":\"Share\",\"root\":\"\",\"domain\":\"domain.com\",\"user\":\"domain.com\\\\User1\",\"password\":\"***removed***\"})"}
{"reqId":"VIPWeSHsaEiBwKxZOQlw","level":0,"time":"2019-05-14T18:40:09+00:00","remoteAddr":"192.168.1.2","user":"User1","app":"smb","method":"GET","url":"\/remote.php\/dav\/uploads\/User1\/web-file-upload-901639628a9eda6f95d484a2f666ff5f-1557859208198","message":"using Icewind\\SMB\\Native\\NativeShare for the connection"}
{"reqId":"VIPWeSHsaEiBwKxZOQlw","level":0,"time":"2019-05-14T18:40:09+00:00","remoteAddr":"192.168.1.2","user":"User1","app":"smb","method":"GET","url":"\/remote.php\/dav\/uploads\/User1\/web-file-upload-901639628a9eda6f95d484a2f666ff5f-1557859208198","message":"leave: __construct, getId:smb::domain.com\\User1@fileserver.domain.com\/\/Share\/\/"}
{"reqId":"VIPWeSHsaEiBwKxZOQlw","level":0,"time":"2019-05-14T18:40:09+00:00","remoteAddr":"192.168.1.2","user":"User1","app":"OC\\Files\\Cache\\Scanner","method":"GET","url":"\/remote.php\/dav\/uploads\/User1\/web-file-upload-901639628a9eda6f95d484a2f666ff5f-1557859208198","message":"!!! Path 'uploads' is not accessible or present !!!"}
{"reqId":"VIPWeSHsaEiBwKxZOQlw","level":3,"time":"2019-05-14T18:40:09+00:00","remoteAddr":"192.168.1.2","user":"User1","app":"remote","method":"GET","url":"\/remote.php\/dav\/uploads\/User1\/web-file-upload-901639628a9eda6f95d484a2f666ff5f-1557859208198","message":"Exception: {\"Exception\":\"Error\",\"Message\":\"Call to a member function getPath() on boolean\",\"Code\":0,\"Trace\":\"#0 \\\/var\\\/www\\\/owncloud\\\/apps\\\/dav\\\/lib\\\/Connector\\\/Sabre\\\/Directory.php(88): OCA\\\\DAV\\\\Connector\\\\Sabre\\\\Node->__construct(Object(OC\\\\Files\\\\View), false, NULL)\\n#1 \\\/var\\\/www\\\/owncloud\\\/apps\\\/dav\\\/lib\\\/Upload\\\/UploadHome.php(97): OCA\\\\DAV\\\\Connector\\\\Sabre\\\\Directory->__construct(Object(OC\\\\Files\\\\View), false)\\n#2 \\\/var\\\/www\\\/owncloud\\\/apps\\\/dav\\\/lib\\\/Upload\\\/UploadHome.php(52): OCA\\\\DAV\\\\Upload\\\\UploadHome->impl()\\n#3 \\\/var\\\/www\\\/owncloud\\\/apps\\\/dav\\\/lib\\\/Upload\\\/UploadHome.php(62): OCA\\\\DAV\\\\Upload\\\\UploadHome->getChild('web-file-upload...')\\n#4 \\\/var\\\/www\\\/owncloud\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Tree.php(105): OCA\\\\DAV\\\\Upload\\\\UploadHome->childExists('web-file-upload...')\\n#5 \\\/var\\\/www\\\/owncloud\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAVACL\\\/Plugin.php(878): Sabre\\\\DAV\\\\Tree->nodeExists('uploads\\\/User1....')\\n#6 [internal function]: Sabre\\\\DAVACL\\\\Plugin->beforeMethod(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#7 \\\/var\\\/www\\\/owncloud\\\/lib\\\/composer\\\/sabre\\\/event\\\/lib\\\/EventEmitterTrait.php(105): call_user_func_array(Array, Array)\\n#8 \\\/var\\\/www\\\/owncloud\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(466): Sabre\\\\Event\\\\EventEmitter->emit('beforeMethod', Array)\\n#9 \\\/var\\\/www\\\/owncloud\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(254): Sabre\\\\DAV\\\\Server->invokeMethod(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#10 \\\/var\\\/www\\\/owncloud\\\/apps\\\/dav\\\/lib\\\/Server.php(288): Sabre\\\\DAV\\\\Server->exec()\\n#11 \\\/var\\\/www\\\/owncloud\\\/apps\\\/dav\\\/appinfo\\\/v2\\\/remote.php(31): OCA\\\\DAV\\\\Server->exec()\\n#12 \\\/var\\\/www\\\/owncloud\\\/remote.php(165): require_once('\\\/var\\\/www\\\/ownclo...')\\n#13 {main}\",\"File\":\"\\\/var\\\/www\\\/owncloud\\\/apps\\\/dav\\\/lib\\\/Connector\\\/Sabre\\\/Node.php\",\"Line\":86}"}
```

**Comment by individual-it** [comment]
@fsutil could you please try with the fresh version 10.2

**Comment by fsutil** [comment]
> @fsutil could you please try with the fresh version 10.2

I removed the workaround in config/config.php ('dav.chunk_base_dir' => '/tmp/uploads',) and tested after upgrading to 10.2. It acts the same way: files larger than 10 MB don't upload. If I add the workaround back in, I can upload files larger than 10 MB.

**Comment by skshetry** [comment]
@fsutil, I couldn't reproduce it locally. I did as what we mentioned above:

1. Add `SMB` as external storage
2. Set `0 B` on `Settings -> User Authentication -> Special Attributes -> Quota Default`.
3. Sync users.
4. Login as any one ldap user, and upload inside the mapped external storage.

I was able to upload files (larger or even smaller than 10 MB). I also tried this on `SFTP`.  I tried this on `master` though.

**Comment by skshetry** [comment]
@fsutil, I tried this also on the current `10.2` release, but could not reproduce.

**Comment by fsutil** [comment]
Hmmm... I’ll have to try again on a new test OwnCloud box.  It was happening on both our test and production instances of OwnCloud, but they would be very similar as far as configuration and updates installed. (I’m currently at a conference, so it’ll be a while before I test this.)

**Comment by fsutil** [comment]
I built a new test ownCloud 10.2 box with Ubuntu 18.04. I haven't been able to reproduce my problem on this new test install, though I did find a few differences:

**Code in new config.php file that wasn't in previous config.php file**
I tried copying this code onto the old test box and still had the same upload issue.

'apps_paths' =>
  array (
    0 =>
    array (
      'path' => '/var/www/owncloud/apps',
      'url' => '/apps',
      'writable' => false,
    ),
    1 =>
    array (
      'path' => '/var/www/owncloud/apps-external',
      'url' => '/apps-external',
      'writable' => true,
    ),
  ),

**Differences in file permissions**
I believe these differences are probably from running the script to set strong directory permissions at some point in the past. [https://doc.owncloud.org/server/10.1/admin_manual/installation/installation_wizard.html#setting-strong-directory-permissions](https://doc.owncloud.org/server/10.1/admin_manual/installation/installation_wizard.html#setting-strong-directory-permissions)
_Old Server:_
drwxr-x---  29 www-data www-data   4096 May 22 16:12 apps
drwxr-xr-x   2 root     root       4096 May  2  2018 apps2
drwxr-xr-x   2 www-data www-data   4096 May 19 02:19 apps-external
drwxr-x---   2 root     www-data   4096 Apr 25  2018 assets
-rw-r--r--   1 root     root       8859 May 16 15:17 AUTHORS
-rw-r--r--   1 root     root     111501 May 16 15:17 CHANGELOG.md
drwxr-x---   2 www-data www-data   4096 Jun 10 20:16 config
-rw-r--r--   1 root     root       4371 May 16 15:17 console.php
-rw-r--r--   1 root     root      34520 May 16 15:17 COPYING
drwxr-x---  16 root     www-data   4096 May 22 16:10 core
-rw-r--r--   1 root     root       5033 May 16 15:17 cron.php
drwxr-x--- 432 www-data www-data  20480 May 22 16:13 data
-rw-r--r--   1 root     root      31203 May 16 15:17 db_structure.xml
drwxr-x---   2 root     www-data   4096 Apr 25  2018 etc
-rw-r--r--   1 root     root        179 May 16 15:17 index.html
-rw-r--r--   1 root     root       3678 May 16 15:17 index.php
drwxr-x---   6 root     www-data   4096 May 22 16:10 lib
-rwxr-xr-x   1 root     root        283 May 16 15:17 occ
drwxr-xr-x   2 root     root       4096 May 22 16:10 ocm-provider
drwxr-x---   2 root     www-data   4096 May 22 16:10 ocs
drwxr-x---   2 root     www-data   4096 May 22 16:10 ocs-provider
-rw-r--r--   1 root     root       3216 May 16 15:17 public.php
-rw-r--r--   1 root     root       2159 May 16 15:17 README.md
-rw-r--r--   1 root     root       5488 May 16 15:17 remote.php
drwxr-x---   4 root     www-data   4096 May  2  2018 resources
-rw-r--r--   1 root     root         26 May 16 15:17 robots.txt
drwxr-x---  12 root     www-data   4096 May 22 16:10 settings
-rw-r--r--   1 root     root       2059 May 16 15:17 status.php
drwxr-x---   4 root     www-data   4096 Mar 28  2018 themes
drwxr-x---   6 www-data www-data   4096 May 22 16:10 updater
-rw-r--r--   1 root     root        280 May 16 15:17 version.php


_New Test Server:_
drwxr-xr-x 22 www-data www-data   4096 Jun 10 17:14 apps
drwxr-xr-x  3 www-data www-data   4096 Jun 10 18:21 apps-external
drwxr-xr-x  2 www-data www-data   4096 May 21 14:36 assets
-rw-r--r--  1 root     root       8859 May 16 15:17 AUTHORS
-rw-r--r--  1 root     root     111501 May 16 15:17 CHANGELOG.md
drwxrwxr-x  2 www-data www-data   4096 Jun 10 20:35 config
-rw-r--r--  1 root     root       4371 May 16 15:17 console.php
-rw-r--r--  1 root     root      34520 May 16 15:17 COPYING
drwxr-xr-x 16 root     root       4096 Jun 10 17:14 core
-rw-r--r--  1 root     root       5033 May 16 15:17 cron.php
drwxrwx---  8 www-data www-data   4096 Jun 10 18:49 data
-rw-r--r--  1 root     root      31203 May 16 15:17 db_structure.xml
drwxr-xr-x  2 root     root       4096 May 21 14:36 etc
-rw-r--r--  1 root     root        179 May 16 15:17 index.html
-rw-r--r--  1 root     root       3678 May 16 15:17 index.php
drwxr-xr-x  6 root     root       4096 Jun 10 17:14 lib
-rwxr-xr-x  1 root     root        283 May 16 15:17 occ
drwxr-xr-x  2 root     root       4096 Jun 10 17:14 ocm-provider
drwxr-xr-x  2 root     root       4096 Jun 10 17:14 ocs
drwxr-xr-x  2 root     root       4096 Jun 10 17:14 ocs-provider
-rw-r--r--  1 root     root       3216 May 16 15:17 public.php
-rw-r--r--  1 root     root       2159 May 16 15:17 README.md
-rw-r--r--  1 root     root       5488 May 16 15:17 remote.php
drwxr-xr-x  4 root     root       4096 Jun 10 17:14 resources
-rw-r--r--  1 root     root         26 May 16 15:17 robots.txt
drwxr-xr-x 12 root     root       4096 Jun 10 17:14 settings
-rw-r--r--  1 root     root       2059 May 16 15:17 status.php
drwxr-xr-x  6 root     root       4096 Jun 10 17:14 updater
-rw-r--r--  1 root     root        280 May 16 15:17 version.php

Any more ideas or things to check?
